### ISSUE TITLE
Add a CLI arg to open a new OS window in the CWD

### ISSUE THREAD

**Issue by ThinkChaos** [kovidgoyal/kitty#7539]
**Is your feature request related to a problem? Please describe.**

I'd like to bind my main terminal shortcut to something that opens a shell in the currently focused shell's working directory.
I don't want to have to use a different shortcut when Kitty is focused.

Right now I'm achieving that by using [swaycwd](https://gitlab.com/cab404/swaycwd) to find the directory to use based on child processes of the focused Sway window: in short it iterates the process tree trying to find a shell and use that working dir.
With Kitty's single instance mode, the shell it finds isn't necessarily the focused one since all windows share a PID. To workaround that I'm using the window's title instead of hunting for a shell. It's a bit hacky and doesn't work in all cases.

**Describe the solution you'd like**

Since Kitty knows which shell is focused, it could support using that when opening a new OS window, with fallback to `.`.
Basically a way to achieve `launch --cwd=current` or `new_os_window_with_cwd` without enabling remote control (assuming I understand the docs for those correctly).
I think the easiest way to opt in to the new behavior would be through a CLI arg, maybe re-use the API from `launch`:  
`--directory=current`.

**Describe alternatives you've considered**

The script as mentioned above: works most of the time, but not ideal.
Enabling remote control: I'd rather not open that can of worms.
Using Kitty builtin tabs/windows: I like my tiling manager setup better.

**Additional context**
N/A

---

Thanks for the software, I use it every day!

**Comment by kovidgoyal** [comment]
remote control is precisely for this. If you think its a can of worms, I
suggest you dont use sway either as it also uses a socket for IPC just
like kitty remote control.

**Comment by ThinkChaos** [comment]
Thanks for the reply.

I'm sorry my message came of as dismissive of remote control, it was not my intention. It's a useful feature that I just personally don't have a need for, with the exception of this feature, so I'd rather keep it disabled to avoid increasing my attack surface.  
I don't think this is news for you, but compared to Sway, it works over SSH, so could potentially allow spreading from a server to my local machine. And can be used by a program to opportunistically escalate privileges: run a command in a TTY that has a valid sudo ticket.
I guess I could use [Fine grained permissions for remote control](https://sw.kovidgoyal.net/kitty/remote-control/#fine-grained-permissions-for-remote-control) to restrict what remote control can do, but that's still a lot more opportunity for bugs compared to not enabling the feature.

I think that's a fair objective assessment, and the fact the feature includes some ways to lock it down implies you agree there's security consequences to enabling it.
With that in mind do you still think this specific feature is not worth making accessible separately?
Would you be open to merging a PR for this?

**Comment by kovidgoyal** [comment]
allow_remote_control socket-only


Means exactly the same attack surface as sway.

**Comment by ThinkChaos** [comment]
Sway doesn't let you send input to windows.  
With kitty you could wait for a window to have a title matching `^sudo ` and then send it text like `sudo whatever\n` to elevate privileges since the sudo ticket would still be valid and no password would be asked.

Anyways thanks for the responses, I'll just live with my script.

**Comment by kovidgoyal** [comment]
On Sun, Jun 16, 2024 at 03:25:17PM -0700, ThinkChaos wrote:
> Sway doesn't let you send input to windows.

I'll leave you with this:
https://github.com/swaywm/sway/issues/1779

**Comment by kovidgoyal** [comment]
Oh and just for completeness, with this commit: https://github.com/kovidgoyal/kitty/commit/9be6b9c3743f6915d82dd7630c9bd0f714dc52e9

you can do what you want with:

```sh
echo 'launch --allow-remote-control --type=background kitten @ launch --cwd=current' | kitty --single-instance --session -
```

this will allow you to use remote control via kitty --single-instance without turning on remote control globally.

**Comment by ThinkChaos** [comment]
Awesome thanks for the thorough reply and quick solution!
